### PR TITLE
[iPhoneX] Converts the status bar to work with iPhone X

### PR DESCRIPTION
**Pull request by orta**
Moves the status bar from each individual ARNAvigationController (e.g. per tab) to the root view controller. 

This means the nav controller's view is now accurately set to include the status bar, which should kill a bunch of hard to diagnose "why doesn't the title look right" bugs we've had with random view controllers and especially artsy views.

#trivial because I already have a entry for this

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m [hunk 1]
```diff
@@ -131,6 +134,16 @@ - (void)viewDidLoad
     self.view.backgroundColor = [UIColor whiteColor];
     self.selectedTabIndex = -1;
 
+    _statusBarView = [[UIView alloc] init];
+    _statusBarView.backgroundColor = UIColor.blackColor;
+
+    [self.view addSubview:_statusBarView];
+
+    _statusBarVerticalConstraint = [_statusBarView constrainHeight:@"20"];
```
Comment (ashfurrow) [inline Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m]: Should this be `statusBarHeight` instead of a hard coded 20?

Comment (orta) [reply Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m]: No, we can't know that at init time, this is just a generic default

Comment (ashfurrow) [reply Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m]: Makes sense! Want to add a comment to that effect? LGTM.

**Pull request merged**